File: spring-utils/src/main/java/chatox/platform/exception/metadata/MetadataEnhancedException.java
package chatox.platform.exception.metadata;

import lombok.Getter;

public class MetadataEnhancedException extends RuntimeException {
    @Getter
    private ExceptionMetadata metadata;

    public MetadataEnhancedException(ExceptionMetadata exceptionMetadata) {
        this.metadata = exceptionMetadata;
    }

    public MetadataEnhancedException(String message, ExceptionMetadata exceptionMetadata) {
        super(message);
        this.metadata = exceptionMetadata;
    }

    public MetadataEnhancedException(String message, Throwable cause, ExceptionMetadata exceptionMetadata) {
        super(message, cause);
        this.metadata = exceptionMetadata;
    }
}

File: chat/src/main/kotlin/chatox/chat/exception/metadata/MetadataEnhancedException.kt
package chatox.chat.exception.metadata

open class MetadataEnhancedException : RuntimeException {
    var metadata: ExceptionMetadata

    constructor(metadata: ExceptionMetadata) : super() {
        this.metadata = metadata
    }

    constructor(message: String?, metadata: ExceptionMetadata) : super(message) {
        this.metadata = metadata
    }
}
